### DISCUSSION TITLE
Special Handling for Domains

### DISCUSSION THREAD

**Discussion by questor** [go-shiori/shiori#404] [Feature requests]
I really like the approach to generate a single-html-page for archival, but I would love to have special handling configurable for special domains, two examples:
- for github-links I would love to automatically clone the repos (which is a rabbit hole on it's own, because how to maintain them)
- for youtube-links it would be great to download the video automatically

I understand that you don't want to include non-go-packages (like for example youtube-dl), so I think there are multiple approaches this could be handled:
- some sort of plugin-system which can be added and configured per usecase written in go
- something generic that you can filter for domains and execute user-defined system-commands (which might have security implications depending on how it's implemented and if it's a public server or not)

there are other implications aswell, for example how to display pages when the user clicked on such a archived link (plugin could provide a generic page customized to this specific link like having a generic page which displays the captured movie) or how to generate an overview picture which can be shown to the user. so I think this needs more thinking about usecases, but would be great to have one solution for ALL bookmarks (like youtube-links, github and so on) to make them available offline.

before diving into finding solutions it would be good to know if something like that is seen in scope for the project and if yes, what are thoughts and preferences...

**Comment by fmartingr** [comment]
Hey @questor, thanks for the suggestions!

I've given thought to this while being an user (and now even more being a collaborator) and I'm unsure where the line resides. All that will be very complex to integrate within Shiori itself because of the simplistic approach, and I'm still trying to understand how far from bookmark manager to archival tool shiori can/needs to be.

In my case, archival of text is enough (not sure how well github repos are achived at the moment, but READMEs would be fine for me) and for videos I use another solution, because it is now only downloading them, storing, playing, qualities... everything is a rabbit hole on its own.

The plugin system is a good idea. I started considering it for features really out of scope of Shiori so people can just communicate with it via RPC or something like that to accomplish more complex things, but this is just an idea right now.

While investigating tools for archival I came across [ArchiveBox](https://github.com/ArchiveBox/ArchiveBox) which may be more suited to what you're looking for, maybe? Take a look and let me know.

**Comment by questor** [comment]
Hi!
don't get me wrong, it's also a totally valid response to say (because of all the rabbit holes) it's not in scope of the project. Plugin-System would be cool so that it could be handled as a separate project. Will also take a look at ArchiveBox, thanks!
